Common/Brokerages/KrakenBrokerageModel.cs

```diff
@@ -0,0 +1,174 @@
+using QuantConnect.Securities;
+using QuantConnect.Orders.Fills;
+using QuantConnect.Orders.Fees;
+using System.Linq;
+
+namespace QuantConnect.Brokerages
+{
+    /// <summary>
+    /// Provides Kraken specific properties
+    /// </summary>
+    public class KrakenBrokerageModel : DefaultBrokerageModel
+    {
+        private static BrokerageMessageEvent _message = new BrokerageMessageEvent(BrokerageMessageType.Warning, 0, "Brokerage does not support update. You must cancel and re-create instead.");
+
+        // https://support.Kraken.com/customer/portal/articles/2725970-trading-rules
+        private static readonly Dictionary<string, decimal> MinimumOrderSizes = new Dictionary<string, decimal>
+        {
+            { "BTCUSD", 0.001m },
+            { "BTCEUR", 0.001m },
+            { "BTCGBP", 0.001m },
+
+            { "BCHUSD", 0.01m },
+            { "BCHEUR", 0.01m },
+            { "BCHBTC", 0.01m },
+
+            { "ETHUSD", 0.01m },
+            { "ETHEUR", 0.01m },
+            { "ETHGBP", 0.01m },
+            { "ETHBTC", 0.01m },
+
+            { "LTCUSD", 0.1m },
+            { "LTCEUR", 0.1m },
+            { "LTCGBP", 0.1m },
+            { "LTCBTC", 0.1m }
+        };
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="KrakenBrokerageModel"/> class
+        /// </summary>
+        /// <param name="accountType">The type of account to be modelled, defaults to
+        /// <see cref="AccountType.Cash"/></param>
+        public KrakenBrokerageModel(AccountType accountType = AccountType.Cash)
+            : base(accountType)
+        {
+            //! Kraken support margin trading but it's disabled for now
+            if (accountType == AccountType.Margin)
+            {
+                throw new Exception("The Kraken brokerage does not currently support Margin trading.");
+            }
+        }
+
+        /// <summary>
+        /// Kraken global leverage rule
+        /// </summary>
+        /// <param name="security"></param>
+        /// <returns></returns>
+        public override decimal GetLeverage(Security security)
+        {
+            return 1m;
+        }
+
+        /// <summary>
+        /// Provides Kraken fee model
+        /// </summary>
+        /// <param name="security"></param>
+        /// <returns></returns>
+        public override IFeeModel GetFeeModel(Security security)
+        {
+            return new KrakenFeeModel();
+        }
+
+        /// <summary>
+        /// Kraken does no support update of orders
+        /// </summary>
+        /// <param name="security"></param>
+        /// <param name="order"></param>
+        /// <param name="request"></param>
+        /// <param name="message"></param>
+        /// <returns></returns>
+        public override bool CanUpdateOrder(Security security, Order order, UpdateOrderRequest request, out BrokerageMessageEvent message)
+        {
+            message = _message;
+            return false;
```

Comment:
This is a good reason to make the `KrakenBrokerage.UpdateOrder` method `throw new NotSupportedException` -- since the infrastructure will block the update call anyway because this model says it can never be updated.

Reply:
Already removed code from `UpdateOrder` and made it throw `NotSupportedException`